Workload tests adapted to centralized result logging

martin-mat opened the pull request:
## Description
Workload tests adapted to cenralized result logging

## Issues:
Refs: #1861 

## How has this been tested:
 - [ ] Covered by existing integration testing
 - [ ] Added integration testing to cover
 - [ ] Verified all A/C passes
     * [ ] develop
     * [ ] master
     * [ ] tag/other branch
 - [ ] Test environment
    * [ ] Shared Packet K8s cluster
    * [ ] New Packet K8s cluster
    * [ ] Kind cluster
 - [ ] Have not tested

## Types of changes:
 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [ ] New feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
 - [ ] Documentation update

## Checklist:
**Documentation**
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] No updates required.

**Code Review**
- [ ] Does the test handle fatal exceptions, ie. rescue block

**Issue**
- [ ] Tasks in issue are checked off

martin-mat commented:
spec test result the same as for main branch (still some failures but those don't not seem to be caused by this change).
Ready for review.

agentpoyo commented:
> spec test result the same as for main branch (still some failures but those don't not seem to be caused by this change). Ready for review.

We'll try to pull down and push up to verify github actions with no issues and manual testing prior to merging.

martin-mat commented on src/tasks/workload/5g_validator.cr:
```diff
@@ -17,10 +17,7 @@ end
-    task_start_time = Time.utc
-    testsuite_task = "smf_upf_core_validator"
-    Log.for(testsuite_task).info { "Starting test" }
+  CNFManager::Task.task_runner(args, task: t) do |args, config|
```
start time of every test case moved to the task runner
https://github.com/cnti-testcatalog/testsuite/blob/main/src/tasks/utils/task.cr#L113

and reported along with test case result the same way as before the changes 
https://github.com/cnti-testcatalog/testsuite/blob/main/src/tasks/utils/task.cr#L121

I'm not aware that test case duration/end time is recorded anywhere. But possible to introduce easily if needed.

martin-mat commented:
rebased

HashNuke commented:
Just tried running the build for this. There are some tests that are failing on repeated attempts.

https://github.com/cnti-testcatalog/testsuite/actions/runs/8232211049

martin-mat commented:
fixed.
@HashNuke in the link provided I saw following failures of the spec tests:
curl, helm, operator_test, points , validate_config

I fixed reasons for: helm, operator_test, points , validate_config.
Failing "curl" specs: I don't know, I think it is not related to this change. For me it is failing even on main branch.

Can you please re-run github actions? Thanks.

HashNuke commented:
* LGTM. Merged this.
* Build is passing - https://github.com/cnti-testcatalog/testsuite/actions/runs/8342423721
